apps/passport-client/components/screens/ImportBackupScreen.tsx

```diff
@@ -0,0 +1,386 @@
+          // the number of PCDs in each folder, with `NoFolderSymbol` used for
+          // PCDs belonging to no folder.
+          const pcdFolders: Record<string | symbol, number> =
+            mergeablePcds.reduce((folders, pcd) => {
+              const folder =
+                parsedCollection.getFolderOfPCD(pcd.id) ?? NoFolderSymbol;
+              if (folder in folders) {
+                folders[folder]++;
+              } else {
+                folders[folder] = 1;
+              }
+              return folders;
+            }, {});
+
+          // The set of folders that the user has chosen to import.
+          let selectedFolders: Set<string | symbol>;
+
+          if (importState.state === "initial") {
+            // By default all folders are selected.
+            selectedFolders = new Set([
+              NoFolderSymbol,
+              ...Object.keys(pcdFolders)
+            ]);
+          } else {
+            // Otherwise, make sure previously selected folders are still
+            // valid given current parsed file contents.
+            selectedFolders = importState.selectedFolders;
+            for (const folder of selectedFolders) {
+              // Do we have a selected folder that doesn't exist any more
+              // for some reason?
+              if (!(folder in pcdFolders)) {
+                selectedFolders.delete(folder);
+              }
+            }
+          }
+
+          setImportState({
+            state: "valid-file-selected",
+            collection: parsedCollection,
+            mergeablePcdIds: new Set(mergeablePcds.map((pcd) => pcd.id)),
+            selectedFolders,
+            // Check which of the mergeable PCDs are in selected folders, and
+            // create a set of their IDs
+            selectedPcdIds: new Set(
+              mergeablePcds
+                .filter((pcd) => {
+                  const folder =
+                    parsedCollection.getFolderOfPCD(pcd.id) ?? NoFolderSymbol;
+                  return selectedFolders.has(folder);
+                })
+                .map((pcd) => pcd.id)
+            ),
+            folderCounts: pcdFolders
+          });
+        }
+      }
+    })();
+  }, [filesContent, importState, existingPcdCollection]);
+
+  // When the user selects or de-selects a folder for inclusion in the merge
+  const toggleFolder = useCallback(
+    (folder: string | symbol) => {
+      if (importState.state === "valid-file-selected") {
+        const { selectedFolders } = importState;
+        if (selectedFolders.has(folder) && selectedFolders.size > 1) {
+          selectedFolders.delete(folder);
+        } else {
+          selectedFolders.add(folder);
+        }
+        setImportState({ ...importState, selectedFolders });
+      }
+    },
+    [importState]
+  );
+
+  return (
+    <>
+      <MaybeModal />
+      <AppContainer bg="gray">
+        <ScreenNavigation label={"Home"} to="/"></ScreenNavigation>
+        <Container>
```

Comment:
This is the sort of thing that should be standardized and not rely on someone remembering to do it when implementing a new screen. #1346